iothub/service/src/Common/Exceptions/DeviceMessageLockLostException.cs

```diff
@@ -14,6 +14,11 @@ public DeviceMessageLockLostException(string message)
         {
         }
 
+        public DeviceMessageLockLostException(ErrorCode code, string message)
```

Comment:
this should have been an overload with `ErrorCode code, string message, Exception innerException = null`, correct?

Reply:
It would be good to keep all exception classes consistent though this class never had innerException in the parameters list.

Reply:
I am seeing tons if inconsistencies between the exception classes. They're all over the place. I did not add innerException here because non of the methods expose it. So left it out. It should probably be there, but then I'll have to investigate all the other exception classes and make them consistent. Since the change is addressing a specific bug, and since we're working on track2, I thought I would make minimal changes to fix the issue. Let me know if you guys disagree.